python/cudf/cudf/core/dataframe.py

```diff
@@ -2576,6 +2576,64 @@ def insert(self, loc, name, value, nan_as_null=None):
 
         self._data.insert(name, value, loc=loc)
 
+    def diff(self, periods=1, axis=0):
+        """
+        Calculates the difference of a Dataframe element compared with
+        another element in the Dataframe, treating each column independently
```

Comment:
I think `pydocstyle` may complain with this formatting? Was still trying to understand if the first line of the docstring should be single sentence and end with period `.` cc @bdice

Reply:
We don't enforce pydocstyle yet but it's somewhere on our roadmap. Pydocstyle would say that the first line of a docstring should be a short summary written in the imperative mood and avoid wrapping to a second line (see [D400, D401, D415](http://www.pydocstyle.org/en/6.1.1/error_codes.html)). This should be followed by a blank line, then the rest of the docstring contents. However, this is often hard to write in practice, and I'd say consistency with pandas should be our priority.

For comparison, [pandas writes](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.diff.html):
```
First discrete difference of element.

Calculates the difference of a Dataframe element compared with another element in the Dataframe (default is element in previous row).
```

Note the first sentence isn't in imperative mood, so pydocstyle wouldn't pass on that. Again, I view consistency with pandas as our top priority for docstrings.